frontend/src/utils/glueSerializer.js

```diff
@@ -1,6 +1,89 @@
 export const glueSerializer = tables => {
   const result = { databases: [] };
 
+  const ARRAYSTRUCTPREFIX = "array<struct<";
+  const STRUCTPREFIX = "struct<";
+  const SCHEMA_INVALID = "Column schema is not valid";
+  const ALLOWEDTYPES = [
+    "bigint",
+    "char",
+    "double",
+    "float",
+    "int",
+    "smallint",
+    "string",
+    "tinyint",
+    "varchar"
+  ];
+
+  const columnMapper = c => {
+    let prefix,
+      suffixLength,
+      resultType = c.Type;
+
+    if (c.Type.startsWith(ARRAYSTRUCTPREFIX)) {
+      resultType = "array<struct>";
+      prefix = ARRAYSTRUCTPREFIX;
+      suffixLength = 2;
+    } else if (c.Type.startsWith(STRUCTPREFIX)) {
+      resultType = "struct";
+      prefix = STRUCTPREFIX;
+      suffixLength = 1;
+    }
+
+    const result = {
+      name: c.Name,
+      type: resultType,
+      canBeIdentifier: ALLOWEDTYPES.includes(resultType)
+    };
+
+    if (prefix) {
+      if (c.Type.slice(-suffixLength) !== ">".repeat(suffixLength))
+        throw new Error(SCHEMA_INVALID);
+
+      result.children = [];
+      let current = c.Type.substr(prefix.length).slice(0, -suffixLength);
+
+      while (true) {
+        if (current.length === 0) break;
+        const name = current.substr(0, current.indexOf(":"));
+        const rest = current.substr(name.length + 1);
+        const isStruct = rest.startsWith(STRUCTPREFIX);
+        const isStructArray = rest.startsWith(ARRAYSTRUCTPREFIX);
+
+        if (isStruct || isStructArray) {
+          const prefix = isStruct ? STRUCTPREFIX : ARRAYSTRUCTPREFIX;
+          let nOpenedTags = isStruct ? 1 : 2;
+          let endIndex = -1;
+          const structSchema = [...rest.substr(prefix.length)];
+          structSchema.some((c, i) => {
+            if (c === "<") nOpenedTags++;
```

Comment:
not too sure about callback passed to `some` mutating vars from the outer scope. doesn't seem like a good approach from an functional perspective (not that were doing FP in this project!!!)

Reply:
I think you're right. Moved to for, which makes it actually more readable